R/Rules-methods.R

```diff
@@ -744,35 +730,50 @@ setMethod("maxDose",
               return(ret)
           })
 
+# nolint end
 
-## --------------------------------------------------
-## The maximum allowable number of dose levels method
-## --------------------------------------------------
-
-##' @describeIn maxDose Determine the maximum possible next dose based on
-##' maximum dose levels to increment for the next dose
-##'
-##' @example examples/Rules-method-maxDose-IncrementsNumDoseLevels.R
-setMethod("maxDose",
-          signature=
-            signature(increments="IncrementsNumDoseLevels",
-                      data="Data"),
-          def=
-            function(increments, data, ...){
-              ## determine what was the level of the last dose
-              lastDoseLevel <- tail(data@xLevel, 1)
-
-              ## determine the maximum next dose level
-              maxNextDoseLevel <- min(length(data@doseGrid),
-                                      lastDoseLevel + increments@maxLevels)
-
-              ## so the maximum next dose is
-              ret <- data@doseGrid[maxNextDoseLevel]
-
-              return(ret)
-            })
+# maxDose-IncrementsNumDoseLevels ----
 
+#' @rdname maxDose
+#' 
+#' @description Increments control based on number of dose levels
+#'   Increment rule to determine the maximum possible next dose based on
+#'   maximum dose levels to increment for the next dose. 
+#'   Increment rule can be applied to last dose or maximum dose given so far.
+#'
+#' @aliases maxDose-IncrementsNumDoseLevels
+#' @example examples/Rules-method-maxDose-IncrementsNumDoseLevels.R
+#' @export
+#' 
+setMethod(
+  "maxDose",
+  signature = signature(
+    increments = "IncrementsNumDoseLevels",
+    data = "Data"
+  ),
+  definition  = function(increments, data, ...) {
+    # Determine what is the basis level for increment,
+    # i.e. the last dose or the max dose applied.
+    basisDoseLevel <- ifelse(
```

Comment:
use snake case for internal variables, so here
```suggestion
    basis_dose_level <- ifelse(
```